src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1302,9 +1491,15 @@ protected TaskWrapper getNextTask() throws InterruptedException {
               }
             }
           } else {
+            if (queuePriority == TaskPriority.High) {
```

Comment:
This, and the added section below is where we notify when there are no high priority tasks ready to run.  You can see the queue must check which priority it is before it can run the notify logic